project_template/inc/az/exampleshortname/operations.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/** @file operations.h */
+#pragma once
+#include <stdint.h>
+#include <az/exampleshortname/export.h>

```

Comment:
It seems worth checking this file in instead of generating it.

Reply:
it may want to live in some core library. If we have specific export headers for each client library (for example to allow building some as static and some as dynamic) then it could make sense to have cmake generate the export headers.

Reply:
After chatting offline, this should be as simple as
```
// project_template/src/az/export.h
#if !defined(BUILDING_DLL)
#define AZ_EXPORT
#elif defined(_MSC_VER)
#define AZ_EXPORT __declspec(dllexport)
#else
#define AZ_EXPORT __attribute__((visibility("default")))
#endif
```
This file won't be installed; it will only be consumed in implementation files and any exported definitions should be marked with `AZ_EXPORT`. The public header files should all be completely un-annotated and should not include this file.